akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -439,8 +440,10 @@ private[client] object NewHostConnectionPool {
 
             emitRequest(newRequest)
           }
-          def close(): Unit = {
-            requestOut.complete()
+          def close(force: Boolean): Unit = {
+            if (force) requestOut.fail(new IllegalStateException("pool slot shutdown") with NoStackTrace)
+            else requestOut.complete()
```

Comment:
:+1: seems premature to just 'complete' it if the stream has not ended 'naturally'